### ISSUE TITLE
Character freezes when using gestures

### ISSUE THREAD

**Issue by Pegwald** [aura-project/aura#159]
I am suddenly unable to spam /dance2. Nothing happens when trying to use any of them and the character will freeze for a short amount of time.

**Comment by exectails** [comment]
When was the last time you updated? The ops changed lately.

**Comment by Pegwald** [comment]
Just hours ago, but I can try updating it again

**Comment by exectails** [comment]
I just tried using gestures, works fine for me o.o The op change was some time last week I think.

**Comment by Pegwald** [comment]
hmm. When I ran the channel I noticed there was some error with Malcom, resulting in him missing from his shop. Like. He took a vacation or something.

Didn't get no error this time, however.

**Comment by Pegwald** [comment]
... But back to the case; I updated and I still can't use gestures. Do I troubleshoot it by just deleting my Aura folder?

**Comment by exectails** [comment]
You did recompile, yes?

**Comment by Pegwald** [comment]
... I did something at least. I'll check if it works now

![image](https://cloud.githubusercontent.com/assets/13089450/12001399/385e4fa4-aae8-11e5-9983-3de6068006ea.png)

**Comment by Pegwald** [comment]
I did nothing. It's still the same *face->table*

**Comment by exectails** [comment]
So... *did* you compile after updating?

**Comment by Pegwald** [comment]
I have no idea what that means, to be honest, so I highly doubt it :D

**Comment by exectails** [comment]
It means opening the Aura.sln in Visual Studio and building the solution.

**Comment by Pegwald** [comment]
oooh... I don't have Visual studio, I think o-o'' What is that even... and Aura.sln? Omg, I must be the worst noob in the community for not knowing anything about this. Pls don't get frustrated qwq''

**Comment by exectails** [comment]
... o.o Aura you using Aura Frontend or something?

**Comment by Pegwald** [comment]
Yes :D
I've always been using frontend

**Comment by exectails** [comment]
Alright, Frontend should *technically* do that for you. But maybe it made a mistake during the update or something. Could you check if the following looks the same in your local Op.cs file?
https://github.com/aura-project/aura/blob/48c20dc9af490ec2ce1d2ddc70aa94a6cc3b1993/src/Mabi/Network/Op.cs#L112

**Comment by Pegwald** [comment]
Looks the same. L 112 Right? It's identical

**Comment by exectails** [comment]
Let's see what else I changed recently... are live updates for upgrade effects working for you? Like, when you level up from 9 to 10, does a level 10 enchant bonus take effect without reqeuipping?

**Comment by Pegwald** [comment]
That one will be tricky to check, but I will try

**Comment by Pegwald** [comment]
... Is there a gain exp command? ^^

**Comment by Pegwald** [comment]
nvm, did it the hard way and the results are positive. I do not need to reequip an enchanted item to gain enchant bonuses from level requirements.

**Comment by exectails** [comment]
This is really weird. You click on a gesture's Use button and nothing happens? The client only freezes for a moment? Wait... when did you update the client last?

**Comment by Pegwald** [comment]
... it has been like... uhh... maybe I should patch it? ^^ ''

**Comment by exectails** [comment]
Yea, that might help xD"

**Comment by Pegwald** [comment]
Hmm... But it doesn't really explain why others can't use their gestures... Do they also have to patch?

**Comment by exectails** [comment]
That's very likely, yes. The ops tell the servers and clients how to interpret a certain message they received. Aura was updated for NA217, where the gesture op changed. If you don't use NA217+, the server won't understand the request from the client to use a gesture, and the clients won't understand the command that creature XY is now using a gesture, because the ops mean something else to them.

That's like me suddenly deciding to call Aura "Boobloo", and earasing Aura from my memory, without telling (updating) you, you'd be thoroughly confused as to wth I'm talking about, and we'd have trouble talking about the same thing.

Bottom line: keep both Aura *and* the client updated at all times.

**Comment by Pegwald** [comment]
Ah, I understand. I will do that and remind the others on my server to do the same ^^

> That's like me suddenly deciding to call Aura "Boobloo"

I really approve of this. 10/10 xD

**Comment by Pegwald** [comment]
And it's working again. Thanks Exec, you're a true hero and a genius ^^
